### DISCUSSION TITLE
pip install needed

### DISCUSSION THREAD

**Discussion by gloa7** [AbdBarho/stable-diffusion-webui-docker#720] [General]
I went through the base setup instructions on the wiki and ran into an issue where inside the container it could not find the right version or it at all 'typing-extensions'
Underneath this value that I have I hope highlighted I put in the following command which succeeded in building:
`pip install typing-extensions --upgrade`

Edit: Adding link for reference:
https://github.com/AbdBarho/stable-diffusion-webui-docker/blob/master/services/AUTOMATIC1111/Dockerfile#L34

**Comment by Sbaero** [comment]
In docker desktop you can go to the exec tab and type out your pip commands.
You can also ls to see where you are and cd through the debian file system.

Alternatively, you can create a file /data/config/comfy/startup.sh and write the commands you wanted to execute above the line starting comfy (you can also add startup flags here).

Here's mine:
python -m pip install sageattention
python -u main.py --listen --port 7860 --use-sage-attention

This works because the author adds whatever you put in this file to the ROOT of execution in entrypoint.sh and it seems to execute before the dockerfile instructions.
